### PR TITLE
Pcap perf/v5

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/7155

Replaces / builds on #10635

### CODE REVIEW

File: src/source-pcap-file.c [hunk 1]
```diff
@@ -136,10 +137,32 @@ void TmModuleDecodePcapFileRegister (void)
...
+        if (ParseSizeStringU32(str, &value) < 0) {
+            SCLogWarning("failed to parse pcap-file.buffer-size %s", str);
+        }
+        if (value > 4096 && value <= PCAP_FILE_BUFFER_SIZE_MAX) {
+            SCLogInfo("Pcap-file will use %d buffer size", (int)value);
+            pcap_g.read_buffer_size = (int)value;
+        } else {
+            SCLogWarning("pcap-file.buffer-size value of %u is invalid. Valid range is 4096-%d",
+                    value, PCAP_FILE_BUFFER_SIZE_MAX);
+        }
```
Comment (jasonish) [inline src/source-pcap-file.c:163]:
```
./src/suricata --pcap-file-buffer-size 4096
```
leads to warning:
```
Warning: pcap: pcap-file.buffer-size value of 4096 is invalid. Valid range is 4096-67108864 [PcapFileGlobalInit:source-pcap-file.c:161]
```

**Review by jasonish — CHANGES_REQUESTED** [review]
Off by one, otherwise looks good when fixup commit merged.

File: src/source-pcap-file.c (hunk 1 above)
Comment (victorjulien) [reply src/source-pcap-file.c:163]: Fixed

**Comment by victorjulien** [comment]
replaced by #11464

**Pull request closed without merging**